### ISSUE TITLE
[Feature] Nuke Password (Password to delete database)

### ISSUE THREAD

**Issue by fux0r2009** [leonlatsch/Photok#84]
**Is your feature request related to a problem? Please describe.**
In some circumstances deleting database needs to be stealthy 

**Describe the solution you'd like**
The best way to silently delete a database is by entering a password giving the impression you are trying to unlock the database when in fact you are deleting it!

**Additional context**
It would be highly recommended to replace the deleted database with a new fake one with a predetermined password and with some random data or just empty to make the process stealthy.

**Comment by VioletFalcon** [comment]
I think just giving support for [Ripple](https://f-droid.org/packages/info.guardianproject.ripple/) is a better idea.

**Comment by leonlatsch** [comment]
@VioletFalcon good idea, I dont know it yet, but will have a look into it.

**Comment by fux0r2009** [comment]
@VioletFalcon Yes that would be good but Ripple hasn't been updated for years now and even if it still works it's not guaranteed to still work in future android releases so having an In-app solution is always better in this case!

**Comment by VioletFalcon** [comment]
@fux0r2009 Well, probably you're right, because anyways official F-Droid client has the feature to uninstall any app you want with the panic trigger.

**Comment by leonlatsch** [comment]
@VioletFalcon Uninstalling the app will always delete the whole database, since backups of the internal files are turned of.

**Comment by leonlatsch** [comment]
@VioletFalcon @fux0r2009 I dont really know ripple. Is it still known and used.
If so, we can add ripple support additionally.

**Comment by VioletFalcon** [comment]
> @VioletFalcon @fux0r2009 I dont really know ripple. Is it still known and used?
> If so, we can add ripple support additionally.

Idk, the maintainer is a F-Droid contributor, sure they can give a better response.
As I said, official F-Droid client already has the option to uninstall Photok + data with the panic trigger so, not really needed IMO

**Comment by leonlatsch** [comment]
> > @VioletFalcon @fux0r2009 I dont really know ripple. Is it still known and used?
> > If so, we can add ripple support additionally.
> 
> Idk, the maintainer is a F-Droid contributor, sure they can give a better response.
> As I said, official F-Droid client already has the option to uninstall Photok + data with the panic trigger so, not really needed IMO

Maybe, but Photok is also available on Gplay and so on. I think, I will give it a try.

**Comment by leonlatsch** [comment]
[This](https://github.com/SilenceIM/Silence/blob/8b2d61161716dcae33c7ae2fd9540931b632030a/src/org/smssecure/smssecure/PanicResponderActivity.java) looks pretty easy to implement and supports many panic buttons. Maybe I will do something like this in addition.

**Comment by fux0r2009** [comment]
> [This](https://github.com/SilenceIM/Silence/blob/8b2d61161716dcae33c7ae2fd9540931b632030a/src/org/smssecure/smssecure/PanicResponderActivity.java) looks pretty easy to implement and supports many panic buttons. Maybe I will do something like this in addition.

Yeah sure why not, If it's easy to implement it won't hurt thank you @leonlatsch for your work as always :)

**Comment by leonlatsch** [comment]
@fux0r2009 @VioletFalcon Do you guys think support for panic buttons is enough?
I don't know if I like the nuke password.
Maybe I could add a button on the lock page to reset the app. But I already had this dilemma before release. Someone could just reset your app without the password. Which is not nice. BUT, they can do it anyway, because they could uninstall the app, soooo.

Whats your opinion?

**Comment by fux0r2009** [comment]
@leonlatsch Yes the panic button support is nice to have but please don't give the option to put a button that could reset the app without a password as this would be catastrophic for the Non-saved data!
